api/event_listener.go

```diff
@@ -0,0 +1,55 @@
+package api
+
+import (
+	"github.com/cskr/pubsub"
+	"github.com/mesg-foundation/core/event"
+)
+
+// EventFilter store fileds for matching events.
+type EventFilter struct {
+	Key string
+}
+
+// Match matches event.
+func (f *EventFilter) Match(e *event.Event) bool {
+	return f == nil || f.Key == "" || f.Key == e.Key
+}
+
+// EventListener provides functionalities to listen MESG events.
+type EventListener struct {
+	C chan *event.Event
```

Comment:
if this is exposed it should have a better name, something like `eventC`

Reply:
C is fine. the struct name (`EventListener`) gives the information about what type of channel it is.
If you want to more clarity in the code you can use longer var names .
```
eventListener  := NewEventListener()
<-eventListenr.C
```

or shorter if it's clear from context.

Example with `C` naming https://golang.org/pkg/time/#Timer